### PR TITLE
Classes in python

### PR DESCRIPTION
**Pull request by zsajjan**

### CODE REVIEW

File: general/lessons/classes/README.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+# Classes in Python
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: it should be obvious that it's in python

File: general/lessons/classes/README.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+>Classes are essentially a user-defined data type. Classes have a few components:
+
+* Initializers
+* Variables
+* Methods
+
+In order to better understand classes, we’ll look at it through a simple example. Imagine a class as a blueprint for a dog.
+
+
+
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: blank lines are unnecessary

File: general/lessons/classes/README.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+----
+## Class Variables
+In order to make our dog with our blueprint, we’ll need to start off by knowing different characteristics of the dog, like fur color, eye color, and age. These attributes can be stored as variables, and are called class variables. 
+
+
+
+
+    class Dog: ← This line signifies that we are
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: put this on a line that doesn't look like code

File: general/lessons/classes/README.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+In order to make our dog with our blueprint, we’ll need to start off by knowing different characteristics of the dog, like fur color, eye color, and age. These attributes can be stored as variables, and are called class variables. 
+
+
+
+
+    class Dog: ← This line signifies that we are 
+    going to start defining a class called Dog
+      fur_color = "golden" ← These lines are where Class Variables are defined.
+    eye_color = "blue"    
+    age = 3
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: 4 space indentation for everything inside the class

File: general/lessons/classes/README.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+----
+## Instances
+
+Now that we have a class (although it is a bit basic right now), we can initialize it.
+Think of this as making your dog!
+The syntax to initialize a class is as follows:
+
+    my_dog = Dog()
+                   
+    Think of this as defining a variable called  my_dog, which is an instance of the class Dog
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: this line is not code

File: general/lessons/classes/README.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+Now that we have our instance, what can we do with it?
+For starters, let’s see if we can print our dog’s age! The code to do this would look like this:
+
+    print my_dog.age
+
+If we want to change our dog’s age, or other characteristic, we would use the following code:
+
+my_dog.age = 4
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: this line is code

File: general/lessons/classes/README.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+The syntax for creating a class with an initializer is as follows:
+
+    class Dog:
+    fur_color = "golden" ← Class variables (always the same, shared by all instances)
+    eye_color = "blue"
+    age = 3
+
+    def __init__(self, name): ← Initializer (these variables are unique to each instance of the class)
+    """the self parameter means that the function concerns other elements of the class, and the name parameter is the user input for what the dog’s name should be"""
+    self.name = name
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: indent

File: general/lessons/classes/README.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+The syntax for creating a class with an initializer is as follows:
+
+    class Dog:
+    fur_color = "golden" ← Class variables (always the same, shared by all instances)
+    eye_color = "blue"
+    age = 3
+
+    def __init__(self, name): ← Initializer (these variables are unique to each instance of the class)
+    """the self parameter means that the function concerns other elements of the class, and the name parameter is the user input for what the dog’s name should be"""
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: use `#` to comment in python

File: general/lessons/classes/README.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+    age = 3
+
+    def __init__(self, name, pos): ← Initializer (these variables are unique to each instance of the class)
+    self.name = name     
+    self.pos = pos           
+
+    def walk(self, dist): ← Class method (functions inside the class Dog that change the variable pos)
+    self.pos += dist   
+
+    my_dog = Dog("Freddy", 0)
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: make it Margy

File: general/lessons/classes/README.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+    my_dog.walk(5)
+    print my_dog.pos ← Returns 5
+
+# Subclasses
+---
+What if we wanted to make different breeds of dog?
+For example, say we wanted to make a dalmatian class.
+In order to do this, we have two options:
+Create a whole new class from scratch
+Create a class based on the Dog() class and change the elements to fit a dalmatian
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: make these 2 options bullet points

File: general/lessons/classes/README.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+    pos += dist   
+
+    class Dalmatian(Dog): ← Child class of Dog
+    fur_color = "spotted"
+    eye_color = "black" ← Only unique elements need to be specified
+
+    my_dalmatian = Dalmation("Fido", 1)
+    print my_dalmatian.age ← The child inherits the age variable from the parent class
+    my_dalmatian.walk(3) ← The child inherits the walk method from the parent class as well!
+    print my_dalmatian.pos ← Returns 4
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: indent code

File: general/lessons/classes/README.md [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+
```
Comment (SMerchant1678) [inline general/lessons/classes/README.md]: remove unnecessary space at the end of the file

File: general/lessons/classes/README.md [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+In order to better understand classes, we’ll look at it through a simple example. Imagine a class as a blueprint for a dog.
+
+----
+## Class Variables
+In order to make our dog with our blueprint, we’ll need to start off by knowing different characteristics of the dog, like fur color, eye color, and age. These attributes can be stored as variables, and are called class variables. 
+
+
+
+
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: unnecessary blank lines

File: general/lessons/classes/README.md [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+    class Dog: #This line signifies that we are
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: put space after hashtag and 2 two spaces before

Comment (nathansolomon1678) [reply general/lessons/classes/README.md]: put comment on one line

File: general/lessons/classes/README.md (hunk 13 above)
Comment (SMerchant1678) [reply general/lessons/classes/README.md]: i agree

File: general/lessons/classes/README.md [hunk 15]
```diff
@@ -0,0 +1,119 @@
...
+    class Dog:
+    fur_color = "golden" #Class variables (always the same, shared by all instances)
+    eye_color = "blue"
+    age = 3
+
+    def __init__(self, name): #Initializer (these variables are unique to each instance of the class)
+    """the self parameter means that the function concerns other elements of the class, and the name parameter is the user input for what the dog’s name should be"""
+    self.name = name       
+
+    my_dog = Dog("Margret")
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: Margy

File: general/lessons/classes/README.md [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+# Subclasses
+---
+What if we wanted to make different breeds of dog?
+For example, say we wanted to make a dalmatian class.
+In order to do this, we have two options:
+* Create a whole new class from scratch
+* Create a class based on the Dog() class and change the elements to fit a dalmatian
+
+As you might have guessed, the second method is much more desirable, and there’s even a word for it: Subclasses
+In the next slide, I’ll go over the syntax for how to make a subclass!
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: remove personal pronouns

File: general/lessons/classes/README.md [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+    fur_color = "golden"
+    eye_color = "blue"
+    age = 3
+
+    def __init__(self, name, pos):
+    self.name = name     
+    self.pos = pos           
+
+    def walk(self, dist):
+    pos += dist
```
Comment (nathansolomon1678) [inline general/lessons/classes/README.md]: indent

File: general/lessons/classes/README.md [hunk 18]
```diff
@@ -0,0 +1,117 @@
...
+
+----
+## Class Variables
+In order to make our dog with our blueprint, we’ll need to start off by knowing different characteristics of the dog, like fur color, eye color, and age. These attributes can be stored as variables, and are called class variables. 
+
+    class Dog: 
+    #This line signifies that we are going to start defining a class called Dog
+    fur_color = "golden" #These lines are where Class Variables are defined.
+    eye_color = "blue"    
+    age = 3
```
Comment (LyraSolomon) [inline general/lessons/classes/README.md]: Indentation

File: general/lessons/classes/README.md [hunk 19]
```diff
@@ -0,0 +1,117 @@
...
+#  Methods
+
+Congrats! You made it this far, but our dog’s not done yet, it needs to walk!
+In order to make our dog be able to walk, the first step would be to have an instance variable for location, but muting this variable will be really clunky if we write a function outside the class.
+This brings us to the next new concept, Class Methods.
+Class methods are functions that are part of the class and change things inside the class.
+In the next side, I’ll present some example code to show you how to make a method to make your dog walk!
+
+    class Dog:
+    	fur_color = "golden"  #Class variables (always the same, shared by all instances)
```
Comment (S-Mackenzie1678) [inline general/lessons/classes/README.md]: "spotted black and brown on white"

File: general/lessons/classes/README.md [hunk 20]
```diff
@@ -0,0 +1,117 @@
...
+    	eye_color = "blue"
```
Comment (S-Mackenzie1678) [inline general/lessons/classes/README.md]: "brown"

File: general/lessons/classes/README.md [hunk 21]
```diff
@@ -0,0 +1,117 @@
...
+    	age = 3
```
Comment (S-Mackenzie1678) [inline general/lessons/classes/README.md]: 16

**Pull request merged**